service/automatic_run_experiment.py

```diff
@@ -62,26 +73,128 @@
+        experiment_requests = experiment_requests[:]  # Don't mutate input.
+        experiment_requests.remove(PAUSE_SERVICE_KEYWORD)
+
+    all_fuzzers = set(fuzzer_utils.get_fuzzer_names())
+    valid = True
+    # Validate format.
+    for request in experiment_requests:
+        if not isinstance(request, dict):
+            logger.error('Request: %s is not a dict.', request)
+            experiment_requests.remove(request)
+            valid = False
+            continue
+
+        if 'experiment' not in request:
+            logger.error('Request: %s does not have field "experiment".',
+                         request)
+            valid = False
+            continue
+
+        experiment = _get_experiment_name(request)
+        if not validate_experiment_name(experiment):
+            valid = False
+            logger.error('Experiment name: %s is not valid.', experiment)
+            # Request isn't so malformed that we can finding other issues.
+            # if present. Don't continue.
+
+        experiment = request['experiment']
+        if not request.get('fuzzers'):
+            logger.error('Request: %s does not specify any fuzzers.', request)
+            valid = False
+            continue
+
+        experiment_fuzzers = request['fuzzers']
+        for fuzzer in experiment_fuzzers:
+            if fuzzer in all_fuzzers:
+                continue
+            # Fuzzer isn't valid.
+            logger.error('Fuzzer: %s in experiment %s is not valid.', fuzzer,
+                         experiment)
+            valid = False
+
+    if not valid:
+        # Don't try the next validation step if the previous failed, we might
+        # exception.
+        return valid
+
+    # Make sure experiment requests have a unique name, we can't run the same
+    # experiment twice.
+    counts = collections.Counter(
+        [request['experiment'] for request in experiment_requests])
+
+    for experiment_name, count in counts.items():
+        if count != 1:
+            logger.error('Experiment: "%s" appears %d times.',
+                         str(experiment_name), count)
+            valid = False
+
+    return valid
+
+
+def run_requested_experiment(dry_run):
+    """Run the oldest requested experiment that hasn't been run yet in
+    experiment-requests.yaml."""
+    requested_experiments = yaml_utils.read(REQUESTED_EXPERIMENTS_PATH)
+
+    # TODO(metzman): Look into supporting benchmarks as an optional parameter so
+    # that people can add fuzzers that don't support everything.
+
+    if PAUSE_SERVICE_KEYWORD in requested_experiments:
+        # Check if automated experiment service is paused.
+        logs.warning('Pause service requested, not running experiment.')
+        return None
+
+    requested_experiment = None
+    for experiment_config in reversed(requested_experiments):
+        experiment_name = _get_experiment_name(experiment_config)
+        experiment_request_is_new = db_utils.query(models.Experiment).filter(
```

Comment:
s/experiment_request_is_new/is_new_experiment to shorten ?

Reply:
Done.